src/@next/components/organisms/CreditCardForm/CreditCardFormContent.tsx

```diff
@@ -0,0 +1,71 @@
+import { compact } from "lodash";
+import React from "react";
+import NumberFormat from "react-number-format";
+
+import { TextField } from "@components/molecules";
+import * as S from "./styles";
+import { CardErrors, PropsWithFormik } from "./types";
+
+const getInputProps = (
+  disabled: boolean,
+  handleChange: (e: React.ChangeEvent) => void
+) => (label: string, errors: CardErrors) => ({
+  customInput: TextField,
+  disabled,
+  errors: compact(errors),
+  label,
+  onChange: handleChange,
+});
+
+export const CreditCardFormContent: React.FC<PropsWithFormik> = ({
+  formRef,
+  cardErrors: {
+    number: cardNumberError,
+    cvv: ccCscError,
+    expirationMonth: expirationMonthError,
+    expirationYear: expirationYearError,
+  },
+  disabled,
+  labelsText: { ccCsc: ccCscText, ccExp: ccExpText, ccNumber: ccNumberText },
+  handleSubmit,
+  handleChange,
+}: PropsWithFormik) => {
+  const basicInputProps = getInputProps(disabled, handleChange);
+
+  return (
+    <S.PaymentForm ref={formRef} onSubmit={handleSubmit}>
+      <S.PaymentInput>
+        <NumberFormat
+          autoFocus
+          autoComplete="cc-number"
+          format="#### #### #### ####"
+          name="ccNumber"
+          {...basicInputProps(ccNumberText, [cardNumberError])}
```

Comment:
As discussed, you will probably need to pass value here from Formik state. It looks like we are using 2 sources of truth here.